src/components/filters/useGroupFilter.js

```diff
@@ -46,9 +46,10 @@ const useGroupFilter = (showNoGroupOption = false) => {
       const firstRequest = !ignore
         ? await getGroups(undefined, { page: 1, per_page: 1 })
         : { total: 0 };
-      const groups = !ignore
-        ? await fetchBatched(getGroups, firstRequest.total)
-        : [];
+      const groups =
+        !ignore && firstRequest.total > 0
+          ? await fetchBatched(getGroups, firstRequest.total)
```

Comment:
The `p-all` library used by the promise queue hook has a `stopOnError` option that we can use. https://www.npmjs.com/package/p-all#stoponerror

Reply:
hm. I just saw the default is true, so it should already do that. <minecraft-villagersound>hm</minecraft-villagersound>

Reply:
@bastilian thanks for having a look!

I was playing a bit with it and I'm able to cancel the rest. Did you mean to also skip the `firstRequest` and do it all in one batch? We would need to somehow pass up the `total` so we know how many requests it will need and since we're doing 2 concurrent requests I'm not sure how it would be done.

I think we could optimize it a different way. Let's make the `firstRequest` get the first 50, if `total > 0` then we can merge it with the `groups` and the batch will be made only if there's more than 50 total systems. Since we fetched the first 50, we can start the `fetchBatched` from 51..100

Reply:
and there won't be a need for any cancellations